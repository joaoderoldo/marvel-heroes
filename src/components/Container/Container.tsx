import clsx from "clsx";
import { FC } from "react";

import { Wrapper } from "./styles";
import { ContainerProps } from "./types";

const Container: FC<ContainerProps> = ({ children, classes }) => {
  return (
    <Wrapper className={clsx(classes)} test-id="container-component">
      {children}
    </Wrapper>
  );
};

export default Container;
