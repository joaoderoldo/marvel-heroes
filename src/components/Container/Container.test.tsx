import { render } from "@testing-library/react";
import Container from "@/components/Container/Container";

describe("Container", () => {
  it("renders Container component correctly and match snapshop", () => {
    const { queryByTestId } = render(<Container>Hello, World!</Container>);
    expect(queryByTestId("container-component")).toMatchSnapshot();
  });

  it("renders Container component correctly and match snapshop with applied class", () => {
    const { queryByTestId } = render(
      <Container classes="foo bar">Hello, World!</Container>
    );
    expect(queryByTestId("container-component")).toMatchSnapshot();
  });

  it("renders children", () => {
    const { getByText } = render(<Container>Hello, World!</Container>);
    expect(getByText("Hello, World!")).toBeInTheDocument();
  });

  it("applies classes", () => {
    const { container } = render(
      <Container classes="foo bar">Hello, World!</Container>
    );
    expect(container.firstChild).toHaveClass("foo bar");
  });
});
