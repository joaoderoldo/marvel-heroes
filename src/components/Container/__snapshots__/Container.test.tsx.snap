// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > renders Container component correctly and match snapshop 1`] = `null`;

exports[`Container > renders Container component correctly and match snapshop with applied class 1`] = `null`;
